### PR TITLE
Migrate anomaly task to use the new Datumaro dataset

### PR DESCRIPTION
**Pull request by gdlg**
### Summary

This PR adds support for the new dataset class with the anomaly task. Note that the format of the data has changed a bit: there is now a global normal/anomalous label available with clear semantic. Compared to the old format, the logic is much simplified as we don’t to distinguish between different data types like Geti or mvtec.

For more details about the format, see the PR in Datumaro: https://github.com/open-edge-platform/datumaro/pull/1901

### Checklist

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have ran e2e tests and there is no issues.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/open-edge-platform/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/open-edge-platform/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/open-edge-platform/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2025 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

### CODE REVIEW

File: library/src/otx/data/dataset/anomaly_new.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+from otx.types.task import OTXTaskType
+
+if TYPE_CHECKING:
+    from datumaro.experimental import Dataset
+
+
+class OTXAnomalyDataset(OTXDataset):
+    """OTXDataset class for anomaly task."""
+
+    def __init__(self, task_type: OTXTaskType, dm_subset: Dataset, **kwargs) -> None:
```
Comment (samet-akcay) [inline library/src/otx/data/dataset/anomaly_new.py:24]: not related to this pr, but looking for your suggestion in general..

here `dm_subset` arg name does not make sense to me as a reader. Is this really a subset? or dataset? Or something else? What are your thoughts?

**Review by samet-akcay — APPROVED** [review]

**Review by ashwinvaidya17 — APPROVED** [review]

File: library/src/otx/data/dataset/anomaly_new.py (hunk 1 above)
Comment (gdlg) [reply library/src/otx/data/dataset/anomaly_new.py:24]: Yes, I agree. I think that we could just rename dm_subset to “dataset”. If you look at the overall code, then this is a subset of the overall dataset but I prefer to name variables based on what they do rather than how it’s used based the rest of the code and I am not sure that it makes to call it a subset.

Besides this class is called OTXAnomalyDataset, and the base class in PyTorch is also called torch.utils.data.Dataset. So it’s not very consistent to call the parameter “subset”.
